android-appcoins-billing/src/main/java/com/appcoins/sdk/billing/helpers/AppcoinsBillingStubHelper.java

```diff
@@ -2,66 +2,201 @@
+  public AppcoinsBillingStubHelper() {
+    this.lockThread = new Object();
+  }
+
   @Override public int isBillingSupported(int apiVersion, String packageName, String type) {
-    return 0;
+
+    if (WalletUtils.hasWalletInstalled()) {
+      try {
+        if (!wasServiceBinded) {
+          createRepository();
+          lockThread.wait();
+        }
+
+        return serviceAppcoinsBilling.isBillingSupported(apiVersion, packageName, type);
+      } catch (RemoteException e) {
+        e.printStackTrace();
+        return ResponseCode.SERVICE_UNAVAILABLE.getValue();
+      } catch (InterruptedException e) {
+        e.printStackTrace();
+        return ResponseCode.SERVICE_UNAVAILABLE.getValue();
+      }
+    } else {
+      return ResponseCode.OK.getValue();
+    }
   }
 
-  @Override
-  public Bundle getSkuDetails(int apiVersion, String packageName, String type, Bundle skusBundle) {
-    ArrayList<String> sku = (ArrayList<String>) skusBundle.get(Utils.GET_SKU_DETAILS_ITEM_LIST);
-    String response = WSServiceController.GetSkuDetailsService(packageName, sku);
+  @Override public synchronized Bundle getSkuDetails(final int apiVersion, final String packageName,
+      final String type, final Bundle skusBundle) {
+
     Bundle responseWs = new Bundle();
-    responseWs.putString(Utils.NO_WALLET_SKU_DETAILS, response);
+    if (WalletUtils.hasWalletInstalled()) {
+      try {
+        if (!wasServiceBinded) {
+          createRepository();
+          synchronized (lockThread) {
+            lockThread.wait();
+          }
+        }
+        return serviceAppcoinsBilling.getSkuDetails(apiVersion, packageName, type, skusBundle);
+      } catch (RemoteException e) {
+        e.printStackTrace();
+        responseWs.putInt(Utils.RESPONSE_CODE, ResponseCode.SERVICE_UNAVAILABLE.getValue());
+      } catch (InterruptedException e) {
+        e.printStackTrace();
+        responseWs.putInt(Utils.RESPONSE_CODE, ResponseCode.SERVICE_UNAVAILABLE.getValue());
+      }
+    } else {
+      ArrayList<String> sku = (ArrayList<String>) skusBundle.get(Utils.GET_SKU_DETAILS_ITEM_LIST);
+      String response = WSServiceController.GetSkuDetailsService(packageName, sku);
+      responseWs.putString(Utils.NO_WALLET_SKU_DETAILS, response);
+    }
     return responseWs;
   }
 
-  @Override public Bundle getBuyIntent(int apiVersion, String packageName, String sku, String type,
+  @Override public synchronized Bundle getBuyIntent(int apiVersion, String packageName, String sku, String type,
       String developerPayload) {
-    WalletUtils.promptToInstallWallet();
-    Bundle response = new Bundle();
-    response.putString(Utils.HAS_WALLET_INSTALLED, "");
-    response.putInt(Utils.RESPONSE_CODE, ResponseCode.OK.getValue());
-    return response;
+    if (WalletUtils.hasWalletInstalled()) {
+      try {
+        if (!wasServiceBinded) {
+          createRepository();
+          synchronized (lockThread) {
+            lockThread.wait();
+          }
+        }
```

Comment:
Isn't this missing the catch block?

catch (RemoteException e) {
        e.printStackTrace();
        responseWs.putInt(Utils.RESPONSE_CODE, ResponseCode.SERVICE_UNAVAILABLE.getValue());
      } catch (InterruptedException e) {
        e.printStackTrace();
        responseWs.putInt(Utils.RESPONSE_CODE, ResponseCode.SERVICE_UNAVAILABLE.getValue());
      }